Feature/2824 date input (#2864)
evidence-dev/evidence: sites/docs/pages/components/date-input/index.md

---
title: Date Input

> sidebar_position: 1

Reviewer: should have a description

queries: 
- orders_by_day.sql
---

> Creates a date picker that can be used to filter a query.

Reviewer: ```suggestion
A date input component allows the user to select a date or a range of dates. The selected dates can be used as inputs to queries or components.
```

Reviewer: This can be the description also

To see how to filter a query using an input component, see [Filters](/core-concepts/filters).

```sql filtered_query
select 
*
from ${orders_by_day}
where day = '${inputs.date_filtering_a_query.value}'
```

<DocTab>
    <div slot='preview'>
            <DateInput
                name=date_filtering_a_query
                data={orders_by_day}
                dates=day
            />

            <BarChart
                data={filtered_query}
                x=day
                y=sales
                y2=num_orders
            />
    </div>

````markdown

```sql filtered_query
select 
    *
from ${orders_by_day}

> where day = '${inputs.range_filtering_a_query.value}'

Reviewer: This example chart looks odd, A barchart with a single day is rarely used.

I think you'd be using this as a one sided filter. 

```suggestion
where day > '${inputs.range_filtering_a_query.value}'
```

Author: oh nice yeah, i like this better

```

<DateInput
    name=range_filtering_a_query
    data={orders_by_day}
    dates=day
/>

<BarChart
    data={filtered_query}
    x=day
    y=sales
    y2=num_orders
/>
````
</DocTab>

## Examples

### Using Date Input from a Query

The Date selected can be accessed through the `inputs.[name].value` 

<DocTab>
    <div slot='preview'>
            <DateInput
                name=date_input_from_query
                data={orders_by_day}
                dates=day
            />

            Date Selected: {inputs.date_input_from_query.value}
    </div>

````markdown
<DateInput
    name=date_range_from_query
    data={orders_by_day}
    dates=day
/>

Date Selected: {inputs.date_input_from_query.value}
````
</DocTab>

### With a Title

<DocTab>
    <div slot='preview'>
            <DateInput
                name=date_range_with_title
                data={orders_by_day}
                dates=day
                title="Select a Date Input"
                range
            />
    </div>

```markdown
<DateInput
    name=date_range_with_title
    data={orders_by_day}
    dates=day
    title="Select a Date Input"

>     range

Reviewer: this example uses a range, but is in the wrong section? remove?
```suggestion
```

                start=2019-01-01
                end=2019-12-31
                range
            />
    </div>

```markdown
<DateInput
    name=manual_date_range
    start=2019-01-01
    end=2019-12-31
    range
/>
```
</DocTab>

## Options

<PropListing 
    name="name"
    description="Name of the DateInput, used to reference the selected values elsewhere as {`{inputs.name.start`} or {`inputs.name.end`}"
    required=true
    options="string"
/>
<PropListing 
    name="data"
    description="Query name, wrapped in curly braces"
    options="query name"
/>
<PropListing 
    name="range"
    description="toggles between a ranged and single input calendar"
    options={["true", "false"]}
    default=false
/>
<PropListing 
    name="dates"
    description="Column name from the query containing Date Input to span"
    options="column name"
/>
<PropListing 
    name="start"
    description="A manually specified start date to use for the range"
    options="string formatted YYYY-MM-DD"
/>
<PropListing 
    name="end"
    description="A manually specified end date to use for the range"
    options="string formatted YYYY-MM-DD"
/>
<PropListing 
    name="title"

>     options="string"

Reviewer: I think it looks odd to just have a floating number in a button. not obvious what to do without a title?

Maybe we can make this title a default?

I know @mcrascal has an different view so happy to discuss

Author: yeah it's definitely missing something, and easily overlooked on its own
